test/steps/v2_steps.py

```diff
@@ -711,12 +883,125 @@ def expect_path(context, path):
+
+
+@when('we make a LookupApplications call with {application_id} and {round}')
+def lookup_applications(context, application_id, round):
+    context.response = context.icl.applications(application_id=int(application_id), round=int(round))
+
+
+@given('a signing account with address "{address}" and mnemonic "{mnemonic}"')
+def signing_account(context, address, mnemonic):
+    context.signing_mnemonic = mnemonic
+
+
+def operation_string_to_enum(operation):
+    if operation == "call":
+        return transaction.OnComplete.NoOpOC
+    elif operation == "create":
+        return transaction.OnComplete.NoOpOC
+    elif operation == "update":
+        return transaction.OnComplete.UpdateApplicationOC
+    elif operation == "optin":
+        return transaction.OnComplete.NoOpOC
+    elif operation == "delete":
+        return transaction.OnComplete.DeleteApplicationOC
+    elif operation == "clear":
+        return transaction.OnComplete.ClearStateOC
+    elif operation == "closeout":
+        return transaction.OnComplete.CloseOutOC
+    else:
+        raise NotImplementedError("no oncomplete enum for operation " + operation)
+
+
+@when('I build an application transaction with operation "{operation:MaybeString}", application-id {application_id}, sender "{sender:MaybeString}", approval-program "{approval_program:MaybeString}", clear-program "{clear_program:MaybeString}", global-bytes {global_bytes}, global-ints {global_ints}, local-bytes {local_bytes}, local-ints {local_ints}, app-args "{app_args:MaybeString}", foreign-apps "{foreign_apps:MaybeString}", app-accounts "{app_accounts:MaybeString}", fee {fee}, first-valid {first_valid}, last-valid {last_valid}, genesis-hash "{genesis_hash:MaybeString}"')
+def build_app_transaction(context, operation, application_id, sender, approval_program, clear_program, global_bytes,
+                          global_ints, local_bytes, local_ints, app_args, foreign_apps, app_accounts, fee,
+                          first_valid, last_valid, genesis_hash):
+    if operation == "none":
+        operation = None
+    else:
+        operation = operation_string_to_enum(operation)
+    if sender == "none":
+        sender = None
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    dir_path = os.path.dirname(os.path.dirname(dir_path))
+    if approval_program == "none":
+        approval_program = None
+    elif approval_program:
+        with open(dir_path + "/test-harness/features/resources/" + approval_program, "rb") as f:
+            approval_program = bytearray(f.read())
+    if clear_program == "none":
+        clear_program = None
+    elif clear_program:
+        with open(dir_path + "/test-harness/features/resources/" + clear_program, "rb") as f:
+            clear_program = bytearray(f.read())
+    if app_args == "none":
+        app_args = None
+    elif app_args:
+        app_args = [bytes(arg, 'ascii') for arg in app_args.split(",")]
```

Comment:
Args is a little more complicated than this. I followed the goal format thinking that we might test string/int/address. 

`str:hello` would be `bytes('hello', 'ascii')`